test/unit/test_provenance.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from pathlib import Path
+from autosubmit.autosubmit import Autosubmit
+from log.log import AutosubmitCritical
+import os
+from unittest.mock import patch
```

Comment:
If you are using VS Code or PyCharm they should have the option to automatically sort the imports (we will enable this in CICD later). In PyCharm there is the option when you right-click a file, "Optimize Imports".

![image](https://github.com/user-attachments/assets/0e618313-e8a9-41a6-a9e3-5e5e704816aa)